pkg/activator/activator.go

```diff
@@ -33,3 +39,27 @@ var (
 		RevisionHeaderNamespace,
 	}
 )
+
+// GetNameAndNamespace If the headers aren't explicitly specified,
+// then decode the revision name and namespace from the Host header.
+func GetNameAndNamespace(name, namespace, host string) (string, string) {
+	if name == "" || namespace == "" {
+		parts := strings.SplitN(host, ".", 4)
+		switch len(parts) {
+		default:
+			return name, namespace
```

Comment:
The ordering is very non-standard. Can we have 2, 3, 4, default?
Better eve, if we have named cases for 2 3 4